overlord/ifacestate/apparmorprompting/prompting_test.go

```diff
@@ -512,35 +531,54 @@ func (s *apparmorpromptingSuite) TestExistingRuleAllowsNewPrompt(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(resp.AllowedPermission, DeepEquals, expectedPermissions)
 
+	s.st.Lock()
+	defer s.st.Unlock()
 	c.Assert(mgr.Stop(), IsNil)
 }
 
 func (s *apparmorpromptingSuite) checkRecordedPromptNotices(c *C, since time.Time, count int) {
-	s.st.Lock()
-	n := s.st.Notices(&state.NoticeFilter{
-		Types: []state.NoticeType{state.InterfacesRequestsPromptNotice},
-		After: since,
-	})
-	s.st.Unlock()
-	c.Check(n, HasLen, count)
+	s.checkRecordedNotices(c, state.InterfacesRequestsPromptNotice, since, count)
 }
 
 func (s *apparmorpromptingSuite) checkRecordedRuleUpdateNotices(c *C, since time.Time, count int) {
-	s.st.Lock()
-	n := s.st.Notices(&state.NoticeFilter{
-		Types: []state.NoticeType{state.InterfacesRequestsRuleUpdateNotice},
-		After: since,
-	})
-	s.st.Unlock()
-	c.Check(n, HasLen, count)
+	s.checkRecordedNotices(c, state.InterfacesRequestsRuleUpdateNotice, since, count)
+}
+
+func (s *apparmorpromptingSuite) checkRecordedNotices(c *C, noticeType state.NoticeType, since time.Time, count int) {
+	timeout := time.Second
+	if count == 0 {
+		// Don't wait so long if we don't expect any notices
+		timeout = 100 * time.Millisecond
+	}
+	receivedCount := 0
+	for receivedCount < count {
+		s.st.Lock()
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		n, err := s.st.WaitNotices(ctx, &state.NoticeFilter{
+			Types: []state.NoticeType{noticeType},
+			After: since,
+		})
+		s.st.Unlock()
+		if count == 0 {
+			c.Assert(err, NotNil)
+			c.Assert(n, HasLen, 0)
+			return
+		}
```

Comment:
I think it is impossible to get in here?

Reply:
The prior `if count == 0` doesn't return, so this should be possible for any call which has `count` set to `0`, as far as I can tell?

Reply:
If `count == 0`, then the loop won't ever be entered. Since the initial `receivedCount < count` won't ever evaluate to true. Since `receivedCount` starts off at `0`.

Reply:
Oh very good catch, that's a bug